File: legacy/_1.9.4/src/main/java/cn/enaium/noexpensive/callback/AnvilTakeOutputCallback.java
package cn.enaium.noexpensive.callback;

import net.legacyfabric.fabric.api.event.Event;
import net.legacyfabric.fabric.api.event.EventFactory;
import net.minecraft.item.ItemStack;

/**
 * @author Enaium
 */
public interface AnvilTakeOutputCallback {
    Event<AnvilTakeOutputCallback> EVENT = EventFactory.createArrayBacked(AnvilTakeOutputCallback.class, (listeners) -> (output) -> {
        for (AnvilTakeOutputCallback listener : listeners) {
            listener.interact(output);
        }
    });

    void interact(ItemStack output);
}

File: legacy/_1.11.2/src/main/kotlin/cn/enaium/noexpensive/callback/AnvilTakeOutputCallback.kt
package cn.enaium.noexpensive.callback

import net.legacyfabric.fabric.api.event.EventFactory
import net.minecraft.item.ItemStack
import java.util.function.Function

/**
 * @author Enaium
 */
fun interface AnvilTakeOutputCallback {
    fun interact(output: ItemStack)

    companion object {
        val EVENT = EventFactory.createArrayBacked(
            AnvilTakeOutputCallback::class.java
        ) { listeners: Array<AnvilTakeOutputCallback> ->
            AnvilTakeOutputCallback { output: ItemStack ->
                for (listener in listeners) {
                    listener.interact(output)
                }
            }
        }
    }
}
